cdap-app-fabric/src/main/java/io/cdap/cdap/gateway/handlers/SourceControlManagementHttpHandler.java

```diff
@@ -58,16 +59,22 @@ public class SourceControlManagementHttpHandler extends AbstractAppFabricHttpHan
   public void setRepository(FullHttpRequest request, HttpResponder responder,
                             @PathParam("namespace-id") String namespaceId) throws Exception {
     RepositoryConfigRequest repoRequest = getRepository(request);
-    if (!repoRequest.shouldValidate()) {
-      if (!repoRequest.isValid()) {
-        throw new BadRequestException(String.format("Invalid repository configuration: %s.", repoRequest));
-      }
-      sourceControlService.setRepository(validateNamespaceId(namespaceId), repoRequest);
-      responder.sendString(HttpResponseStatus.OK, String.format("Updated repository configuration for namespace '%s'.",
-                                                                namespaceId));
+
+    if (repoRequest.shouldValidate()) {
+      // TODO: CDAP-20252, add the validate logic once the SourceControlManager module is ready
+    }
+
+    try {
+      sourceControlService.setRepository(validateNamespaceId(namespaceId), repoRequest.getRepository());
```

Comment:
it's better to validate the namespace earlier. With it down here, it will be possible to run the validate logic on an invalid namespace.

Reply:
updated